### ISSUE TITLE
Could not load gtools_macosx_v3.plugin, error 9999

### ISSUE THREAD

**Issue by tkemeny** [mcaceresb/stata-gtools#73]
Hi there,

I'm trying to use gtools on a Mac and Stata 16.1.   I thought I had a bug, as I cannot get gcollapse to run. I get the following error:

gcollapse (mean) hw=hourwg aw=annwg [iw=wgt], by(czone year)
Could not load plugin: /Users/tk1/Library/Application Support/Stata/ado/plus/g/gtools_macosx_v3.plugin
(error occurred while loading _gtools_internal.ado)
r(9999);

But I think it must have something to do with the installation. Note below: 

. which gtools
/Users/tk1/Library/Application Support/Stata/ado/plus/g/gtools.ado
*! version 1.7.5 18Apr2020 Mauricio Caceres Bravo, mauricio.caceres.bravo@gmail.com
*! Program for managing the gtools package installation

. gtools
Nothing to do. See help gtools or gtools, examples for usage. Version info:
/Users/tk1/Library/Application Support/Stata/ado/plus/g/gtools.ado
*! version 1.7.5 18Apr2020 Mauricio Caceres Bravo, mauricio.caceres.bravo@gmail.com
*! Program for managing the gtools package installation
Could not load plugin: /Users/tk1/Library/Application Support/Stata/ado/plus/g/gtools_macosx_v3.plugin
(error occurred while loading _gtools_internal.ado)
(warning: gtools_plugin internal check failed)

Any thoughts on how to resolve?

Thanks
Tom

**Comment by mcaceresb** [comment]
@tkemeny You can try compiling `gtools` yourself, as described [here](https://gtools.readthedocs.io/en/latest/compiling/index.html). Be sure to uninstall the version you currently have before you try.

**Comment by florianpeters** [comment]
I have the same issue, ie error message 
Could not load plugin: /Users/fpeters/Library/Application Support/Stata/ado/plus/g/gtools_macosx_v3.plugin
(error occurred while loading _gtools_internal.ado)
r(9999);

I'm using  a new MacBook Air (M1, 2020) with an Apple M1 chip and Stata MP 17.0.

The file gtools_macosx_v3.plugin is located in the correct folder but cannot be loaded. 

I tried to self-compile according to your instructions:
git clone https://github.com/mcaceresb/stata-gtools
cd stata-gtools
git submodule update --init --recursive
# Stata 14.1 and later
make clean SPI=3.0 SPIVER=v3
make all   SPI=3.0 SPIVER=v3

But the tests (do build/gtools_tests.do) dont pass.

Any idea what could be the problem?

Thanks in advance!

**Comment by mcaceresb** [comment]
@florianpeters There are a few reasons why the tests might not pass. A better first diagnosis is to run

```
gtools
```

by itself and see if there is an error. Further, you should be sure to install `gtools` from the local folder where you compiled it. If there is no error when you run `gtools` by itself from a Stata prompt, feel free to post the log from the tests to see what went wrong.

**Comment by florianpeters** [comment]
Magic, it worked! Thanks so much! Big improvement in many functions! 

Key is indeed that gtools must be installed from the local folder where you compiled it. So, altogether, in the terminal, I ran: 
git clone https://github.com/mcaceresb/stata-gtools
cd stata-gtools
git submodule update --init --recursive
make clean SPI=3.0 SPIVER=v3
make all SPI=3.0 SPIVER=v3

Then in Stata I ran: 
ssc uninstall gtools
net install gtools from(/path_to_folder_where_gtools_was_compiled/stata-gtools/build) replace
gtools

... and I get:
Nothing to do. See help gtools or gtools, examples for usage. Version info:
.../Library/Application Support/Stata/ado/plus/g/gtools.ado
*! version 1.7.5 18Apr2020 Mauricio Caceres Bravo, mauricio.caceres.bravo@gmail.com
*! Program for managing the gtools package installation
(note: gtools_plugin v1.7.5 successfully loaded)

**Comment by dballaelliott** [comment]
I had the same issue on an M1 and the manual install florian describes above fixed things for me as well! Thanks Florian! 

I might suggest adding a note to this effect to the README in the installation section, as I imagine this will be a common issue for M1 users! Thanks for this great package, Mauricio!

**Comment by jdingel** [comment]
Just ran into this on a new M1 machine. The installation instructions from @florianpeters solved my problem. There is a missing comma in Florian's `net install` line, it should say:
```
 net install gtools, from(/path_to_folder_where_gtools_was_compiled/stata-gtools/build) replace
 ```

**Comment by diegojavierjimenez** [comment]
Same issue (M1 machine, same error), same solution as @florianpeters @jdingel.  Thanks for that!

**Comment by mcaceresb** [comment]
Should be fixed in 6dc0fd6bbcc19e6f41f1095ffeaac39a51b7283f. OSX plugin compilation moved to github actions with latest os version.

**Comment by Nanoochoa** [comment]
I had the same issue today  with an M1 machine. The even with the last version [https://github.com/mcaceresb/stata-gtools/commit/6dc0fd6bbcc19e6f41f1095ffeaac39a51b7283f](url), but the fix here works! :)
